broker-node/app/Http/Controllers/UploadSessionController.php

```diff
@@ -117,22 +117,26 @@ public function update(Request $request, $id)
          * may need to return more stuff from processNewChunk
          */
 
-        $updatedChunk = BrokerNode::processNewChunk($brokerReq);
-
-        if (!is_null($updatedChunk) && is_string($updatedChunk) &&
-            $updatedChunk == 'already attached') {
-            return response('Not Accepted.', 204);
-        } else if (!is_null($updatedChunk)) {
-            $data_map->hooknode_id = $updatedChunk->hookNodeUrl;
-            $data_map->trunkTransaction = $updatedChunk->trunkTransaction;
-            $data_map->branchTransaction = $updatedChunk->branchTransaction;
-            $data_map->address = $shortened_hash;
-            $data_map->message = $message_in_tryte_format;
-            $data_map->chunk = $chunk["data"];
-            $data_map->status = DataMap::status['pending'];
-            $data_map->save();
-
-            return response('Success.', 204);
+        [$res_type, $updatedChunk] = BrokerNode::processNewChunk($brokerReq);
+
+        switch($res_type) {
+            case 'already_attached':
```

Comment:
@rfornea this changes what you did. Can you please make sure I understood what you were doing and changed it correctly?

Reply:
This looks like it will be fine.  As long as it's still returning all the data that's on a chunk, and it looks like it is, we should be good to go.